File: frontend/src/types/task.ts
export type TaskCategory = 'ORIENTATION' | 'TRAINING' | 'DOCUMENTATION' | 'MEETING' | 'SHADOWING' | 'SETUP' | 'OTHER';
export type TaskStatus = 'NOT_STARTED' | 'IN_PROGRESS' | 'COMPLETED' | 'BLOCKED';
export type Priority = 'LOW' | 'MEDIUM' | 'HIGH' | 'URGENT';

export interface TaskEntry {
  id: string;
  userId: string;
  date: string;
  title: string;
  description: string;
  category: TaskCategory;
  status: TaskStatus;
  priority: Priority;
  createdAt: string;
  updatedAt: string;
}

export interface CreateTaskRequest {
  date: string;
  title: string;
  description: string;
  category?: TaskCategory;
  status?: TaskStatus;
  priority?: Priority;
}

export interface UpdateTaskRequest extends CreateTaskRequest {}

export interface TaskFilterParams {
  status?: TaskStatus;
  category?: TaskCategory;
  priority?: Priority;
  dateFrom?: string;
  dateTo?: string;
}

File: backend/src/main/java/com/onboardingdiary/entity/Task.java
package com.onboardingdiary.entity;

import com.onboardingdiary.enums.TaskCategory;
import com.onboardingdiary.enums.TaskPriority;
import com.onboardingdiary.enums.TaskStatus;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "tasks")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private LocalDate date;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    private TaskCategory category;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private TaskStatus status = TaskStatus.NOT_STARTED;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 10)
    @Builder.Default
    private TaskPriority priority = TaskPriority.MEDIUM;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}
